### PR TITLE
PBM-429 Log mongodump progress

### PR DESCRIPTION
**Pull request by dAdAbird**
It's going to look something like that
```
2020/05/06 21:31:12 Backup 2020-05-06T18:31:12Z started on node rs2/localhost:28018
2020-05-06T21:31:14.797+0300	writing admin.system.users to archive on stdout
2020-05-06T21:31:14.799+0300	done dumping admin.system.users (2 documents)
2020-05-06T21:31:14.800+0300	writing admin.system.roles to archive on stdout
2020-05-06T21:31:14.807+0300	done dumping admin.system.roles (1 document)
2020-05-06T21:31:14.807+0300	writing admin.system.version to archive on stdout
2020-05-06T21:31:14.815+0300	done dumping admin.system.version (3 documents)
2020-05-06T21:31:14.816+0300	writing test.testt to archive on stdout
2020-05-06T21:31:14.829+0300	writing test.testt2 to archive on stdout
2020-05-06T21:31:14.829+0300	writing config.cache.chunks.config.system.sessions to archive on stdout
2020-05-06T21:31:14.832+0300	done dumping config.cache.chunks.config.system.sessions (1 document)
2020-05-06T21:31:14.834+0300	writing config.cache.collections to archive on stdout
2020-05-06T21:31:14.835+0300	done dumping config.cache.collections (1 document)
2020/05/06 21:31:24 [##......................]   test.testt  130841/1073901  (12.2%)
2020/05/06 21:31:24 [##########..............]  test.testt2   131370/300000  (43.8%)
2020/05/06 21:31:24 
2020/05/06 21:31:34 [#####...................]   test.testt  249603/1073901  (23.2%)
2020/05/06 21:31:34 [###################.....]  test.testt2   249603/300000  (83.2%)
2020/05/06 21:31:34 
2020/05/06 21:31:37 [########################]  test.testt2  300000/300000  (100.0%)
2020-05-06T21:31:39.363+0300	done dumping test.testt2 (300000 documents)
2020/05/06 21:31:44 [#########...............]  test.testt  433521/1073901  (40.4%)
2020/05/06 21:31:54 [##############..........]  test.testt  656850/1073901  (61.2%)
2020/05/06 21:32:04 [###################.....]  test.testt  893316/1073901  (83.2%)
2020/05/06 21:32:12 [########################]  test.testt  1073901/1073901  (100.0%)
2020-05-06T21:32:12.981+0300	done dumping test.testt (1073901 documents)
2020/05/06 21:32:13 mongodump finished, waiting for the oplog
2020/05/06 21:32:15 Backup 2020-05-06T18:31:12Z finished
```

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | passed |
| mongodb 4.0 | passed |
| mongodb 4.2 | passed |

File: pbm/backup/backup.go [hunk 1]
```diff
@@ -589,8 +588,18 @@ func newDump(curi string, conns int) *mdump {
...
+// "logger" for the mongodup's ProgressManager.
+// need it to be able to write new progress data in a new line
+type progressWriter struct{}
+
+func (*progressWriter) Write(m []byte) (int, error) {
+	log.Printf("%s", m)
+	return len(m), nil
+}
+
 // Write always return 0 as written bytes. Needed to satisfy interface
```
Comment (akira-kurogane) [inline pbm/backup/backup.go:600]: This comment is now non-applicable, right?

Comment (dAdAbird) [reply pbm/backup/backup.go:600]: No, the comment is still applicable. It is related to the WriteTo method. And we still don't know how many bytes were written by the mongodump. We only could know how many bytes were written by the progress manager (logger let's say) of mongodump.

File: pbm/backup/backup.go [hunk 2]
```diff
@@ -603,13 +612,16 @@ func (d *mdump) WriteTo(w io.Writer) (int64, error) {
...
+		ProgressManager: pm,
 	}
 	err := mdump.Init()
 	if err != nil {
 		return 0, errors.Wrap(err, "init")
 	}
-	return 0, errors.Wrap(mdump.Dump(), "make dump")
+	pm.Start()
+	err = mdump.Dump()
+	pm.Stop()
```
Comment (blez) [inline pbm/backup/backup.go:623]: It's not improve anything in this code but `defer pm.Stop()` will look more Go-like and save us from bugs if we will have more code between `pm.Start()` and `pm.Stop()` in future

Comment (dAdAbird) [reply pbm/backup/backup.go:623]: At first, I did with the defer and then didn't quite like how looks. But, yes, maybe you're right.

File: pbm/backup/backup.go [hunk 3]
```diff
@@ -589,8 +588,18 @@ func newDump(curi string, conns int) *mdump {
...
+type progressWriter struct{}
+
+func (*progressWriter) Write(m []byte) (int, error) {
+	log.Printf("%s", m)
+	return len(m), nil
+}
+
 // Write always return 0 as written bytes. Needed to satisfy interface
 func (d *mdump) WriteTo(w io.Writer) (int64, error) {
+	pm := progress.NewBarWriter(&progressWriter{}, time.Second*60, 24, false)
```
[the remark this answers is not in the document]
Comment (dAdAbird) [reply pbm/backup/backup.go:602]: Yes, 10 secs frames were made just to sample a more explicit example.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | passed |
| mongodb 4.2 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | passed |
| mongodb 4.0 | passed |
| mongodb 4.2 | passed |

**Pull request merged**